xdsl/irdl.py

```diff
@@ -1162,8 +1183,8 @@ def irdl_op_definition(cls: type[_OpT]) -> type[_OpT]:
     """Decorator used on classes to define a new operation definition."""
```

Comment:
(Not saying this is a good idea, just something to think about)

Could this function be annotated as returning an IRDLOperation, which would then give us all the typechecking behaviors we want, without having to rewrite the `Operation` in all the dialects?

If so, I would have to think about whether this would make sense or not. I am thinking it probably wouldn't, but it's good to know anyhow.

Reply:
I tried, but I don't think there's a way, but I'm surprised and would be super happy to do it this way!

Reply:
Ah one thing we might want to do in this PR is replace `_OpT` with a new `TypeVar` that's bound to `IRDLOperation`

Reply:
It's done 🙂 line 323